daemon/logger/loggerutils/rotatefilewriter.go

```diff
@@ -33,45 +44,70 @@ func NewRotateFileWriter(logPath string, capacity int64, maxFiles int) (*RotateF
 	}
 
 	return &RotateFileWriter{
-		f:            log,
-		capacity:     capacity,
-		currentSize:  size,
-		maxFiles:     maxFiles,
-		notifyRotate: pubsub.NewPublisher(0, 1),
+		f:             log,
+		buf:           bytes.NewBuffer(nil),
+		capacity:      capacity,
+		currentSize:   size,
+		lastTimestamp: defaultTimestamp,
+		maxFiles:      maxFiles,
+		notifyRotate:  pubsub.NewPublisher(0, 1),
 	}, nil
 }
 
-//WriteLog write log message to File
-func (w *RotateFileWriter) Write(message []byte) (int, error) {
-	w.mu.Lock()
+//WriteLog write log logEntry to File
+func (w *RotateFileWriter) Write(logEntry *jsonlog.JSONLogs) error {
+	w.writerMu.Lock()
 	if w.closed {
-		w.mu.Unlock()
-		return -1, errors.New("cannot write because the output file was closed")
+		w.writerMu.Unlock()
+		return errors.New("cannot write because the output file was closed")
 	}
+	w.writerMu.Unlock()
+
 	if err := w.checkCapacityAndRotate(); err != nil {
-		w.mu.Unlock()
-		return -1, err
+		return err
 	}
 
-	n, err := w.f.Write(message)
+	w.writerMu.Lock()
+	defer w.writerMu.Unlock()
```

Comment:
We don't want to use `defer` in this function since this is in a hot path and `defer` carries with it a significant amount of overhead.